File: app/src/main/java/com/platdmit/simplecloudmanager/app/vm/DomainListViewModel.java
package com.platdmit.simplecloudmanager.app.vm;

import com.platdmit.simplecloudmanager.domain.models.Domain;
import com.platdmit.simplecloudmanager.domain.repo.DomainBaseRepo;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.LiveDataReactiveStreams;
import androidx.lifecycle.MutableLiveData;
import io.reactivex.rxjava3.processors.BehaviorProcessor;

public class DomainListViewModel extends BaseViewModel {
    private static final String TAG = DomainListViewModel.class.getSimpleName();
    private DomainBaseRepo mDomainRepo;

    private LiveData<List<Domain>> mDomains;
    private MutableLiveData<String> mResultMassage = new MutableLiveData<>();

    private final BehaviorProcessor<List<Domain>> mContentProvider = BehaviorProcessor.create();

    public DomainListViewModel(DomainBaseRepo domainBaseRepo) {
        mDomainRepo = domainBaseRepo;

        //Fast fix for prevent overSubscription after resize
        mCompositeDisposable.add( mDomainRepo.getDomains().subscribe(mContentProvider::onNext));

        mDomains = LiveDataReactiveStreams.fromPublisher(mContentProvider);
    }

    public void reloadDomainList() {

    }

    public LiveData<List<Domain>> getDomainsLiveData() {
        return mDomains;
    }

    public MutableLiveData<String> getResultMassage() {
        return mResultMassage;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        mContentProvider.onComplete();
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/app/vm/DomainListViewModel.kt
package com.platdmit.simplecloudmanager.app.vm

import androidx.lifecycle.LiveData
import androidx.lifecycle.LiveDataReactiveStreams
import androidx.lifecycle.MutableLiveData
import com.platdmit.simplecloudmanager.domain.models.Domain
import com.platdmit.simplecloudmanager.domain.repo.DomainBaseRepo
import io.reactivex.rxjava3.processors.BehaviorProcessor

class DomainListViewModel(
        private val mDomainRepo: DomainBaseRepo
) : BaseViewModel() {
    val domainsLiveData: LiveData<List<Domain>>
    val resultMassage = MutableLiveData<String>()
    private val mContentProvider = BehaviorProcessor.create<List<Domain>>()

    init {
        //Fast fix for prevent overSubscription after resize
        mCompositeDisposable.add(mDomainRepo.getDomains().subscribe {mContentProvider.onNext(it)})
        domainsLiveData = LiveDataReactiveStreams.fromPublisher(mContentProvider)
    }

    fun reloadDomainList() {
        mDomainRepo.nextUpdate()
    }

    override fun onCleared() {
        super.onCleared()
        mContentProvider.onComplete()
    }

    companion object {
        private val TAG = DomainListViewModel::class.java.simpleName
    }
}